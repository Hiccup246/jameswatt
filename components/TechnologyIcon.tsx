export default function TechnologyIcon({ icon }: { icon: LinkedIcon }) {
  return (
    <a
      href={icon.link}
      tabIndex={-1}
      aria-label={icon.name}
      title={icon.name}
      className={
        "group relative mt-1 flex h-[86px] w-[86px] items-center justify-center rounded-md border border-lightgrey bg-white duration-300 dark:bg-softwhite"
      }
    >
      <div className="flex h-12 w-12 justify-center">{icon.icon}</div>
      <div
        className={
          "group-hover:animation-play absolute -left-0.5 -top-0.5 -z-10 h-[calc(100%+4px)] h-full w-[calc(100%+4px)] w-full rounded-lg bg-tech-icon-gradient opacity-0 transition-all duration-300 ease-linear group-hover:animate-rotate group-hover:opacity-100 dark:bg-dark-tech-icon-gradient"
        }
      />
    </a>
  );
}
