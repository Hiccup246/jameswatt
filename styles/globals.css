@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  :root {
    --slider-items: 7;
    --slider-item-sets: 2;
    --all-slider-items: calc(var(--slider-items) * var(--slider-item-sets));
    --slider-item-width: 200px;
  }

  html,
  body {
    padding: 0;
    margin: 0;
    font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont,
      "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif,
      "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol",
      "Noto Color Emoji";
    @apply bg-brown;
  }

  @property --gradient-angle {
    syntax: "<angle>";
    inherits: false;
    initial-value: 1turn;
  }
}

@layer utilities {
  .animation-pause {
    animation-play-state: paused;
  }

  .animation-play {
    animation-play-state: running;
  }
}
