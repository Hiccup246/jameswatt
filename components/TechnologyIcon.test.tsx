import { expect, test } from "@jest/globals";
import TechnologyIcon from "./TechnologyIcon";
import renderer from "react-test-renderer";

const mockTechnologyIcon: TechnologyIcon = {
  link: "www.example.com",
  name: "example icon",
  icon: <svg></svg>,
};

test("renders the TechnologyIcon correctly", () => {
  const tree = renderer
    .create(<TechnologyIcon icon={mockTechnologyIcon} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
