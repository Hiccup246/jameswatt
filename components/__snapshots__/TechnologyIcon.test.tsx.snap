// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnologyIcon renders correctly 1`] = `
<div
  className="slide relative h-[100px] w-[200px]"
>
  <div
    className="hover group relative h-[86px] w-[86px] duration-150 hover:-translate-y-0.5 hover:shadow-md"
  >
    <a
      aria-label="example icon"
      className="
                        relative z-20 mt-1 flex h-full w-full items-center justify-center
                        rounded-md border border-lightgrey bg-white group-hover:border-0
                "
      href="www.example.com"
      title="example icon"
    >
      <div
        className="flex h-12 w-12 justify-center"
      >
        <title>
          example icon
        </title>
        <svg />
      </div>
    </a>
    <div
      className="
                    group-hover:animation-play absolute
                    top-0 left-0
                    z-10 h-[calc(100%+2px)]
                    w-[calc(100%+2px)] -translate-x-px -translate-y-px rounded-lg bg-tech-icon-gradient
                    opacity-0 transition-all duration-300 ease-linear
                    group-hover:animate-rotate group-hover:opacity-100
                "
    />
  </div>
</div>
`;
